File: Week1/2_Jewels_And_Stones/Solution.cpp
/*
 * Created on Sat May 02 2020
 *
 * Title: Leetcode - Jewels And Stones
 *
 * Author: Vatsal Mistry
 * Web: mistryvatsal.github.io
 */

#include <iostream>
#include <unordered_set>

using namespace std;

class Solution{
    public:
    unordered_set<char> jewels;
    int count=0;
    int numJewelsInStones(string J, string S) {
        if(J.length() || S.length()) return 0;

        for(auto j: J) {
            jewels.insert(j);
        }
        for(auto s: S) {
            if(jewels.find(s) != jewels.end()) count++;
        }
        return count;
    }
};

File: Week1/2_Jewels_And_Stones/Solution.py
#
# Created on Sat May 02 2020
#
# Title: Leetcode - Jewels And Stones
#
# Author: Vatsal Mistry
# Web: mistryvatsal.github.io
#

class Solution:
    def numJewelsInStonesS(self, J, S):
        if len(J)==0 or len(S) == 0: return 0
        count=0
        for s in S:
            if s in J:
                count += 1
        return count

